assets/js/upload.js

```diff
@@ -171,9 +171,127 @@ function setupImageUpload() {
     window.removeEventListener('beforeunload', beforeUnload);
   }
 
+  function createTagError(message) {
+    const buttonAfter = $('#tagsinput-save');
+    const errorElement = makeEl('span', { className: 'help-block tag-error' });
+
+    errorElement.innerText = message;
+    buttonAfter.parentElement.insertBefore(errorElement, buttonAfter);
+  }
+
+  function clearTagErrors() {
+    const tagErrorElements = $$('.tag-error');
+
+    // remove() causes the length to decrease
+    while (tagErrorElements.length > 0) {
+      tagErrorElements[0].remove();
+    }
+  }
+
+  // populate tag error helper bars as necessary
+  // return true if all checks pass
+  // return false if any check fails
+  function validateTags() {
+    const tags = $$('.tag');
+
+    if (tags.length === 0) {
+      createTagError('Tag input must contain at least 3 tags');
+      return false;
+    }
+
+    const tagsArr = [];
+
+    for (const i in tags) {
+      let tag = tags[i].innerText;
+
+      tag = tag.substring(0, tag.length - 2); // remove " x" from the end
+      tagsArr.push(tag);
+    }
```

Comment:
Instead of this, you should select the tags from the tag input textarea:
```js
    const tags = $('textarea.js-taginput').value.split(',').map(t => t.trim());
```

This is because the "fancy" editor automatically fills tags into the "plain" editor as they are typed, but not the other way around, so this would otherwise break uploading for people who choose to use the "plain" editor.

Also, the upload.js logic is reused for the reverse search page, which does not contain a tag input.
Optimally, you should also check to see if the textarea exists before proceeding to handle that case, and then fail gracefully if it does not.

```suggestion
    const tagInput = $('textarea.js-taginput');
    if (!tagInput) {
      return true;
    }
    
    const tagsArr = tagInput.value.split(',').map(t => t.trim());
```

With this solution the need to duplicate the length check is also removed, so it can be done in one spot.